### ISSUE TITLE
Screen not re-drawn when there is no events pending

### ISSUE THREAD

**Issue by SzymonPobiega** [migueldeicaza/gui.cs#207]
I've created a simple view that is meant to display the `tail` of the application log. There is a background `Task` that looks like this:

```
var writer = Task.Run(async () =>
{
    int lines = 0;
    while (true)
    {
        await Task.Delay(1000);
        Debug.WriteLine("Print");
        Application.MainLoop.Invoke(() =>
        {
            messagesText.Print(new string('d', lines));
            Debug.WriteLine("Invoke Print");
            lines++;
        });
    }
});
```

The print method is as simple as

```
lines.Add(line);
SetNeedsDisplay();
```

What I am observing in the Debug output is "Print" messages are coming regularly but "Invoke Print" is only called when there are any UI events happening e.g. mouse moving over the console window. I looked at the code briefly and I think that the `MainLoop`'s `EventsPending` method might be the reason if the driver blocks until an event happens. 

Is there a way to force UI update from outside that does not depend on the UI events happening?

**Comment by INeedAUniqueUsername** [comment]
It seems like adding a dummy timer slows down the program and prevents it from taking input. I need a way to let the UI update idly while also accepting input in between those updates.

**Comment by INeedAUniqueUsername** [comment]
Related: #155

**Comment by migueldeicaza** [comment]
Is this only happening on Windows?

As for the Windows piece: yes, I believe that is the problem - that I have never implemented Wakeup

**Comment by SzymonPobiega** [comment]
@migueldeicaza I never tried running on different drivers so I don't know. By the way, I love it ❤️

**Comment by tig** [comment]
Fixed in #514
